pkg/reporter/reporter.go

```diff
@@ -156,6 +167,77 @@ func (tr JSONReporter) Publish(r report.Report) error {
 	return err //nolint:wrapcheck
 }
 
+//nolint:nestif
+func (tr GitHubReporter) Publish(r report.Report) error {
+	if r.Violations == nil {
+		r.Violations = []report.Violation{}
+	}
+
+	for _, violation := range r.Violations {
+		_, err := fmt.Fprintf(tr.out,
+			"::%s file=%s,line=%d,col=%d::%s\n",
+			violation.Level,
+			violation.Location.File,
+			violation.Location.Row,
+			violation.Location.Column,
+			fmt.Sprintf("%s. To learn more, see: %s", violation.Description, getDocumentationURL(violation)),
+		)
+		if err != nil {
+			return err //nolint:wrapcheck
+		}
+	}
+
+	pluralScanned := ""
+	if r.Summary.FilesScanned == 0 || r.Summary.FilesScanned > 1 {
+		pluralScanned = "s"
+	}
+
+	// https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-job-summary
+	if summaryFileLoc, ok := os.LookupEnv("GITHUB_STEP_SUMMARY"); ok && summaryFileLoc != "" {
+		summaryFile, err := os.OpenFile(summaryFileLoc, os.O_APPEND|os.O_WRONLY, 0o644)
+		if err != nil {
+			return err //nolint:wrapcheck
+		}
+
+		defer func() {
+			_ = summaryFile.Close()
+		}()
+
+		_, _ = fmt.Fprintf(summaryFile, "### Regal Lint Report\n\n")
```

Comment:
[nit] Does the linter forbid us to just do this?
```suggestion
		fmt.Fprintf(summaryFile, "### Regal Lint Report\n\n")
```

Reply:
I thought it did but it seems not. I'll clean this up!